### PR TITLE
[bugfix] enable depth test for all commands in 3d queue

### PR DESCRIPTION
**Pull request by PatriceJiang**

### CODE REVIEW

File: cocos/renderer/CCRenderer.h [hunk 1]
```diff
@@ -372,6 +377,21 @@ class CC_DLL Renderer
     };
     ScissorState _scissorState;
     
+    enum StateFlag {
+        DEPTH_TEST = 1 << 0,
+        DEPTH_WRITE = 1 << 1,
+        CULL_FACE = 1 << 2,
+    };
```
Comment (minggo) [inline cocos/renderer/CCRenderer.h]: It seems this enum is not needed.

File: cocos/renderer/CCRenderer.h [hunk 2]
```diff
@@ -372,6 +377,21 @@ class CC_DLL Renderer
...
+
+    struct StateBlock{
+        uint32_t modifiedStates = 0;
+        bool depthTest = false;
+        bool depthWrite = false;
+        backend::CullMode  cullMode = backend::CullMode::BACK;
+    };
+
+    std::deque<StateBlock> _stateBlockStack;
```
Comment (minggo) [inline cocos/renderer/CCRenderer.h:386]: Need a stack here?

File: cocos/renderer/CCRenderer.cpp [hunk 3]
```diff
@@ -320,17 +320,27 @@ void Renderer::visitRenderQueue(RenderQueue& queue)
...
     doVisitRenderQueue(queue.getSubQueue(RenderQueue::QUEUE_GROUP::OPAQUE_3D));
+    restoreStateBlock();
     
     //
     //Process 3D Transparent object
     //
+    saveStateBlock(StateFlag::DEPTH_TEST | StateFlag::CULL_FACE | StateFlag::DEPTH_WRITE);
+    setDepthTest(true); //enable depth test in 3D queue by default
+    setDepthWrite(false);
+    setCullMode(backend::CullMode::BACK);
```
Comment (minggo) [inline cocos/renderer/CCRenderer.cpp]: it seems can combine the setting with above codes

File: cocos/renderer/CCRenderer.h (hunk 2 above)
Comment (PatriceJiang) [reply cocos/renderer/CCRenderer.h:386]: GroupCommand can cause a recursive calling, that we need a stack to track states.

File: tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp [hunk 4]
```diff
@@ -1861,7 +1861,7 @@ void UseCaseSprite3D::switchCase()
         circleBack->addChild(circle);
         circle->runAction(RepeatForever::create(RotateBy::create(3, Vec3(0.f, 0.f, 360.f))));
 
-        circleBack->setRotation3D(Vec3(90, 90, 0));
+        circleBack->setRotation3D(Vec3(-90, -90, 0));
```
Comment (minggo) [inline tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp:1864]: Why modify it?

Comment (PatriceJiang) [reply tests/cpp-tests/Classes/Sprite3DTest/Sprite3DTest.cpp:1864]: The sprite is discarded by cullface, rotate to keep front face upside.

**Pull request merged**